core/test/libraries/Oracle.t.sol

```diff
@@ -0,0 +1,349 @@
+// SPDX-License-Identifier: AGPL-3.0-only
+pragma solidity 0.8.17;
+
+import "forge-std/Test.sol";
+
+import {Q16} from "src/libraries/constants/Q.sol";
+
+import "src/libraries/Oracle.sol";
+
+contract LibraryWrapper {
+    function observe(
+        IUniswapV3Pool pool,
+        uint32 target,
+        uint256 seed,
+        int24 tick,
+        uint16 observationIndex,
+        uint16 observationCardinality,
+        uint128 liquidity
+    ) external view returns (int56, uint160) {
+        return Oracle.observe(pool, target, seed, tick, observationIndex, observationCardinality, liquidity);
+    }
+}
+
+contract OracleTest is Test {
+    IUniswapV3Pool constant POOL = IUniswapV3Pool(0x85149247691df622eaF1a8Bd0CaFd40BC45154a9);
+
+    // `POOL`'s oracle is fully populated in this block
+    uint256 constant BLOCK_A = 70_000_000;
+    // `POOL`'s oracle is in the process of increasing cardinality, i.e. some entries have `init = false`
+    uint256 constant BLOCK_B = 10_799_800;
+
+    function setUp() public {
+        vm.createSelectFork(vm.rpcUrl("optimism"));
+        vm.rollFork(BLOCK_A);
+    }
+
+    /*//////////////////////////////////////////////////////////////
+                              COMPARATIVE
+    //////////////////////////////////////////////////////////////*/
+
+    /// forge-config: default.fuzz.runs = 1024
+    function test_comparative_observe(uint32 secondsAgo, uint256 seed, bool initializing) public {
+        if (initializing) vm.rollFork(BLOCK_B);
+        else vm.rollFork(BLOCK_A);
+
+        (, int24 tick, uint16 idx, uint16 cardinality, , , ) = POOL.slot0();
+        uint128 liquidity = POOL.liquidity();
+
+        (uint32 oldest, ) = _oldestAndNewest();
+        secondsAgo = uint32(bound(secondsAgo, 0, block.timestamp - oldest));
+
+        // Uniswap method
+        uint32[] memory secondsAgos = new uint32[](1);
+        secondsAgos[0] = secondsAgo;
+        (int56[] memory tickCumulatives, uint160[] memory secondsPerLiquidityCumulativeX128s) = POOL.observe(
+            secondsAgos
+        );
+        int56 tickCumA = tickCumulatives[0];
+        uint160 liqCumA = secondsPerLiquidityCumulativeX128s[0];
+
+        // Our method
+        (int56 tickCumB, uint160 liqCumB) = Oracle.observe(
+            POOL,
+            uint32(block.timestamp - secondsAgo),
+            seed,
+            tick,
+            idx,
+            cardinality,
+            liquidity
+        );
+
+        // Make sure they're the same
+        assertEq(tickCumB, tickCumA);
+        assertEq(liqCumB, liqCumA);
+    }
+
+    /// forge-config: default.fuzz.runs = 1024
+    function test_comparative_observeOld(uint32 secondsAgo, uint256 seed, bool initializing) public {
```

Comment:
Did you mean to add `assertEq`'s to this test? In the other `_comparative_` tests you had an `assertEq` comparing two values - that's why I'm curious?

Reply:
In this case I'm expecting both methods to revert with code "OLD", instead of expecting both methods to have the same output.